### PR TITLE
Large PR apologies 

### PR DESCRIPTION
**Pull request by Yong-L**
Add webhook validator and duplicate request validator.

Also added database uniqueness and rails uniqueness to the constraint in case of race conditions.

Some refactoring to make idempotency which @danthaman44 made transferrable to our webhooks api

Also added `ENV` changes so it'll be tad bit easier to make webhooks work on your local system. I've already setup the environment variables to staging and prod

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,9 @@
 # or operating system, you probably want to add a global ignore instead:
 #   git config --global core.excludesfile '~/.gitignore_global'
 
+# Ignore env file
+.env
```
[the remark this answers is not in the document]
Comment (Yong-L) [reply .gitignore:8]: I thought exposing out api keys wasn’t best practice which was why I thought we had the google drive link

[the remark this answers is not in the document]
Comment (Yong-L) [reply .gitignore:8]: Only problem I see is running tests on github haha

[the remark this answers is not in the document]
Comment (jtmckibb) [reply .gitignore:8]: As long as we aren't exposing private production keys — we should be fine. The only thing people could potentially do is hit our sandbox account.

[the remark this answers is not in the document]
Comment (jtmckibb) [reply .gitignore:8]: If you think that there is risk in exposing our .env file that's fine. We can keep it in a Google Drive. I trust your judgement here

[the remark this answers is not in the document]
Comment (Yong-L) [reply .gitignore:8]: To keep people in sync of exactly what keys people require, I could push up a .env.example file to point out which keys are mandatory and required for folks to setup their environment

File: app/services/duplicate_request_validator.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Validates idempotency using the ExistingEvent Model
+class DuplicateRequestValidator < BaseService
```
[the remark this answers is not in the document]
Comment (Yong-L) [reply app/services/duplicate_request_validator.rb:4]: Thanks! I'm having fun :)

**Pull request merged**